src/main/java/bio/terra/janitor/app/configuration/JdbcConfiguration.java

```diff
@@ -65,8 +66,11 @@ private void configureDataSource() {
     PoolableConnectionFactory poolableConnectionFactory =
         new PoolableConnectionFactory(connectionFactory, null);
 
+    GenericObjectPoolConfig<PoolableConnection> config = new GenericObjectPoolConfig<>();
```

Comment:
you can add a variable like
private Boolean jmxEnabled = false;

Then JANITOR_DB_JMX_ENABLED becomes an app config that we can easily set.

Reply:
Good idea; could we then default it `true` and make unit tests configure it `false`, which would then make this a test-only change?

Also, can `JanitorJdbcConfiguration` change this, given it would need to be in the base class (`JdbcConfiguration`) to be available in `configureDataSource()`? 

(`JanitorJdbcConfiguration` is annotated with `@ConfigureProperties`, but `JdbcConfiguration` is not)

Reply:
I am not sure if we need Jmx Enabled in 'prod' server or not, and I don't know if we are using that to monitoring things anywhere. So I am not sure what is the default value should be. But default to `true`
 and only `false` in test sounds good.

Sorry, just saw this class is not `@ConfigurationProperties `. So we may need override method someing like `getJmxEnabled()`. 

Actaully, can you try just to put:
`spring.jmx.enabled=false` in `application-test.yml`

This comment is totally optional, I don't think we need jmx, should be fine to just disable it everywhere. 
It is up to you, or see what Will suggests.

Reply:
Actually, adding the parameter in `JdbcConfiguration` works just fine, since `@ConfigureProperties` just uses the setter defined in the base class.  This allows it to work for `STAIRWAY_DB...` too.

Thanks for this suggestion, I prefer not changing the normal behavior given this is working around something we might remove.

Reply:
adding `spring.jmx.enabled=false` in `application-test.yml` will not change normal behaviour. the **application-test** is only used in test.

Reply:
I like Yonghao's suggestion. If we can avoid changing the JdbcConfiguration and control JMX enabled by an existing spring property `spring.jmx.enabled` then not doing a code change for this test sgtm. Still needs your lovely comments in application-test.yml.

Reply:
I had tried `spring.jmx.enabled=false` before, but that still exhibits the problem.  I suspect that this config parameter is controlling how Spring interacts with JMX (i.e. registering it's own components for management), rather than disabling it.

I agree that making this pure config would be the best solution; I'll look a little more to see if this is an option, but will probably time-box this research.